### PR TITLE
Travis CI build.

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Added Travis build script to be able to run basic build for all pull requests.

### CODE REVIEW

**Comment by Tomas-Kraus** [comment]
Please squash it into a single commit. I forget to do it on my side.

**Comment by Tomas-Kraus** [comment]
jsonp did not have travis. I asked eclipse guys to configure it and here is simple job to run at least build and tests with every pull request. There is also Jenkins instance on https://ci.eclipse.org/jsonp/ which can do more complicated tasks (artifact signing/deployment, releases, ...)

**Comment by keilw** [comment]
If there's already a Jenkins per Project, why should we use both? @m0mus has to decide if it makes sense or not.

**Comment by lukasj** [comment]
@keilw It is not possible to have per pull request verification on jenkins directly integrated with github, just clink on 'show all checks' within this PR to see it in action (well, it's possible but quite difficult to set up). OTOH it's much easier to keep release related stuff on the jenkins (due to signing requirements etc). Why do you think they can not co-exist and supplement each other?

**Comment by keilw** [comment]
@lukasj I guess that's a restriction by Eclipse Foundation and the way its Jenkins jobs and pipelines are set up, or the particular Jenkins package it got. I am pretty sure, Jenkins as such can do that, but if only this workaround covers both, then why not. It more or less puts Jenkins in a second class position, but for long term branch control to see if everything remains green it probably can't hurt.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: java
+
+jdk:
+  - oraclejdk8
+  - oraclejdk9
+
+install: true
+
+script:
+  - mvn -U -C -Dnon.final=true clean package
```
Comment (lukasj) [inline .travis.yml]: should be 'verify' instead of 'package'

**Comment by m0mus** [comment]
@keilw i remember that we did some Travis integration in JSONB, but never used it. It's totally OK to use Travis and Eclipse supports it.

**Comment by m0mus** [comment]
@keilw Werner, I actually asked Tomas to add Travis support to JSONP. Lukas well explained the reasons. The idea is to use Travis as a quick pre-commit verification and Eclipse Jenkins to produce nightly builds running full tests suite including TCKs.

**Comment by Tomas-Kraus** [comment]
Yes, verify is there. I also turned on caching in Travis CI to avoid downloading artifacts all the time.

**Comment by Tomas-Kraus** [comment]
...and added master build status into README, it's nice to see it on project page

**Pull request merged**